nemo/collections/cv/examples/mnist_lenet5_image_classification.py

```diff
@@ -0,0 +1,103 @@
+
+import nemo.utils.argparse as nm_argparse
+from nemo.collections.cv.modules.data_layers import MNISTDataLayer
+from nemo.collections.cv.modules.losses import NLLLoss
+from nemo.collections.cv.modules.trainables import LeNet5
+from nemo.core import (
+    DeviceType,
+    EvaluatorCallback,
+    NeuralGraph,
+    NeuralModuleFactory,
+    OperationMode,
+    SimpleLossLoggerCallback,
+)
+from nemo.utils import logging
+
+if __name__ == "__main__":
+    # Create the default parser.
+    parser = argparse.ArgumentParser(parents=[nm_argparse.NemoArgParser()], conflict_handler='resolve')
+    # Parse the arguments
+    args = parser.parse_args()
+
+    # Instantiate Neural Factory.
+    nf = NeuralModuleFactory(local_rank=args.local_rank, placement=DeviceType.GPU)
+
+    # Data layers for training and validation.
+    dl = MNISTDataLayer(height=32, width=32, train=True)
+    dl_e = MNISTDataLayer(height=32, width=32, train=False)
+    # The "model".
+    lenet5 = LeNet5()
+    # Loss.
+    nll_loss = NLLLoss()
+
+    # Create a training graph.
+    with NeuralGraph(operation_mode=OperationMode.training) as training_graph:
+        x, y = dl()
+        p = lenet5(images=x)
+        loss = nll_loss(predictions=p, targets=y)
+        # Set output - that output will be used for training.
+        training_graph.outputs["loss"] = loss
+
+    # Display the graph summmary.
+    logging.info(training_graph.summary())
+
+    # Create a validation graph, starting from the second data layer.
+    with NeuralGraph(operation_mode=OperationMode.evaluation) as evaluation_graph:
+        x, y = dl_e()
+        p = lenet5(images=x)
+        loss_e = nll_loss(predictions=p, targets=y)
+
+    # Display the graph summmary.
+    logging.info(evaluation_graph.summary())
+
+    # Create the callbacks.
+    def eval_loss_per_batch_callback(tensors, global_vars):
+        if "eval_loss" not in global_vars.keys():
+            global_vars["eval_loss"] = []
+        for key, value in tensors.items():
+            if key.startswith("loss"):
+                global_vars["eval_loss"].append(mean(stack(value)))
+
+    def eval_loss_epoch_finished_callback(global_vars):
+        eloss = max(tensor(global_vars["eval_loss"]))
+        logging.info("Evaluation Loss: {0}".format(eloss))
+        return dict({"Evaluation Loss": eloss})
+
+    ecallback = EvaluatorCallback(
+        eval_tensors=[loss_e],
+        user_iter_callback=eval_loss_per_batch_callback,
+        user_epochs_done_callback=eval_loss_epoch_finished_callback,
+        eval_step=100,
+    )
+
+    # SimpleLossLoggerCallback will print loss values to console.
+    callback = SimpleLossLoggerCallback(
+        tensors=[loss], print_func=lambda x: logging.info(f'Training Loss: {str(x[0].item())}')
+    )
+
+    # Invoke the "train" action.
+    nf.train(
+        training_graph=training_graph,
+        callbacks=[callback, ecallback],
+        optimization_params={"num_epochs": 10, "lr": 0.001},
```

Comment:
I guess the same comment for all examples: please take things like num_epochs, lr, etc. from args. With whatever "good" defaults.

Reply:
the same reply, this is an example for dummies, makes no sense to parametrize that more